### PR TITLE
docs: add Claude framework enhancements

### PR DESCRIPTION
**Pull request by david-waltermire**
## Summary

Adds enhancements to the Claude Code framework:

- **Build verification summary format** - Standardized format for reporting build results with specific counts for tests, SpotBugs, PMD, and Checkstyle
- **Unit testing rule** - Standards for what to test (edge cases, business logic) and what not to test (getters/setters, trivial code)
- **Unit test writing skill** - Detailed workflow with edge case checklist, coverage expansion process, and test structure patterns

### `.claude/rules/development-workflow.md`
- Added Build Verification Summary Format section
- Added `unit-test-writing` to project skills list

### `.claude/rules/unit-testing.md` (new)
- Core principles: what to test, what not to test
- Prioritize edge cases over happy paths
- Mandatory coverage evaluation when touching existing code

### `.claude/skills/unit-test-writing.md` (new)
- Edge case checklist (Empty/Null, Boundaries, Invalid, Missing, etc.)
- Coverage expansion workflow (4 steps)
- Test naming conventions
- Test fixture patterns (focused fixtures, helper methods, nested classes)

## Test plan

- [x] Documentation-only changes, no code impact
- [x] Reviewed for consistency with existing framework

## Summary by CodeRabbit

* **Documentation**
  * Added unit testing standards and a mandatory coverage-evaluation workflow to assess and expand test coverage when modifying code.
  * Introduced a new "unit-test-writing" skill with guidance, an edge-case checklist, naming and fixture patterns, and examples.
  * Inserted a Build Verification Summary format (sample, guidelines, failure example) into the development workflow.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

**Comment by david-waltermire** [comment]
@coderabbitai review

File: .claude/rules/development-workflow.md [hunk 1]
```diff
@@ -123,6 +124,36 @@ No Issues → Proceed to Verification
...
+- Use ✅ for passing checks (no blocking errors/violations), ❌ for failures
+- Use ⚠️ for coverage below target (build succeeds but coverage warning)
+- Always report both errors/violations AND warnings for each tool
+- Add brief context for notable items (e.g., "import order fixed")
+- Report failures clearly so they can be addressed before proceeding
+
+**Example with failures:**
+```text
+Build failed:
+- ✅ Tests: 54 passed, 2 failed
```
[the remark this answers is not in the document]
Comment (david-waltermire) [reply .claude/rules/development-workflow.md]: Fixed in commit 8d75d6c10 - changed ✅ to ❌ for the failing tests line. Also made the Build Verification Summary Format a subsection (####) of Phase 5 to clarify its scope.
